File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/TypeTemplateService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbTypeTemplate;
import entity.PageResult;
import entity.PygResult;

import java.security.acl.LastOwnerException;
import java.util.List;
import java.util.Map;

public interface TypeTemplateService {

    /**
     * 展示模板列表
     * @param tbTypeTemplate
     * @param page
     * @param size
     * @return
     */
    public PageResult search(TbTypeTemplate tbTypeTemplate, int page, int size);

    /**
     * 添加模板
     * @param tbTypeTemplate
     * @return
     */
    public void add(TbTypeTemplate tbTypeTemplate);

    /**
     * 根据id读取模板信息
     * @param id
     * @return
     */
    public TbTypeTemplate findById(long id);

    /**
     * 更新模板
     * @param tbTypeTemplate
     */
    public void update(TbTypeTemplate tbTypeTemplate);

    /**
     * 删除或者批量删除
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 获取类型模板下拉列表
     * @return
     */
    public List<Map> typeTemplateOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/typeTemplateService.js
app.service("typeTemplateService",function ($http) {

    this.search = function (page,size,searchTypeTemplate) {
        return $http.post("../typeTemplate/search.do?page="+page+"&size="+size,searchTypeTemplate);
    }
    
    this.save = function (entity) {
        return $http.post("../typeTemplate/add.do",entity);
    }

    this.findTypeTemplateById = function (id) {
        return $http.get("../typeTemplate/findTypeTemplateById.do?id="+id);
    }

    this.update = function (entity) {
        return $http.post("../typeTemplate/update.do",entity);
    }
    this.del = function(ids) {
        return $http.get("../typeTemplate/delete.do?ids="+ids);
    }
})